src/AzureRMTools.ts

```diff
@@ -188,13 +190,34 @@ export class AzureRMTools {
             await this.insertItem(TemplateSectionType.Resources, actionContext);
         });
         registerCommand("azurerm-vscode-tools.resetGlobalState", resetGlobalState);
+
+        // Code action commands
         registerCommand("azurerm-vscode-tools.codeAction.addAllMissingParameters", async (actionContext: IActionContext, source?: vscode.Uri) => {
             await this.addMissingParameters(actionContext, source, false);
         });
         registerCommand("azurerm-vscode-tools.codeAction.addMissingRequiredParameters", async (actionContext: IActionContext, source?: vscode.Uri) => {
             await this.addMissingParameters(actionContext, source, true);
         });
 
+        // Code lens commands
+        registerCommand(
+            "azurerm-vscode-tools.codeLens.gotoParameterValue",
+            async (actionContext: IActionContext, uri: vscode.Uri, param: string) => {
+                let textDocument: vscode.TextDocument = await vscode.workspace.openTextDocument(uri);
+                const editor = await vscode.window.showTextDocument(textDocument);
+                const dp = this.getOpenedDeploymentParameters(uri);
+                if (dp) {
+                    // If the parameter isn't in the param file, show the properties section or beginning
+                    //   of file.
+                    const span = dp.getParameterValue(param)?.value?.span
+                        ?? dp.parametersProperty?.nameValue.span
+                        ?? new Language.Span(0, 0);
+                    const range = getVSCodeRangeFromSpan(dp, span);
+                    editor.selection = new vscode.Selection(range.start, range.end);
+                    editor.revealRange(range);
+                }
+            });
+
```

Comment:
Nit: would it make sense to make this a class method rather than an anonymous function here? Seems like that would be the pattern the rest of the command handlers are following.

Reply:
Will do.  This whole file needs refactoring.